update the release docs in preparation for 2.6.0 release (#1524)
GMLC-TDC/HELICS: CHANGELOG.md

A note on future revisions.
Everything within a major version number should be code compatible (with the exception of experimental interfaces). The most notable example of an experimental interface is the support for multiple source inputs. The APIs to deal with this will change in future minor releases. Everything within a single minor release should be network compatible with other federates on the same minor release number. Compatibility across minor release numbers may be possible in some situations but we are not going to guarantee this as those components are subject to performance improvements and may need to be modified at some point. Patch releases will be limited to bug fixes and other improvements not impacting the public API or network compatibility. Check the [Public API](./docs/Public_API.md) for details on what is included and excluded from the public API and version stability.

## [2.6.0][] ~ 2020-08-14

Bug fixes and major logging update

### Changed

> - the build flag function now returns correct debug or release flags depending on the build

Reviewer: ```suggestion
- The build flag function now returns correct debug or release flags depending on the build
```

- The debug postfix `d` is no longer added to the interface libraries
- Spdlog is now being used for logging inside HELICS and the old logger has been removed this results in fewer thread being generated by HELICS.
- CMake will now error if the install directory is set to the build directory

> - some argument names in the C API have been changed for consistency

Reviewer: ```suggestion
- Some argument names in the C API have been changed for consistency
- Output a more descriptive error message for mismatched data sizes when converting types #1521
```

### Fixed

- String with negative numerical values were not acknowledging the negation Issue #1306

> - config file parsing was not acknowledging "unit" string #1512
> - a performance issue with the tcpss core in some cases has been resolved by setting the no_delay option

Reviewer: ```suggestion
- Config file parsing was not acknowledging "unit" string #1512
- A performance issue with the tcpss and tcp cores in some cases has been resolved by setting the no_delay option
- Inconsistency in type returned by endpoint getType in C++98 API #1523
```

### Added

> - flags for `dumplog`, and `force_logging_flush` were added to the C API
> - added missing C++98 call to `getCurrentTime`
> - added `closeLibrary` function to the C++98 API
> - added a Python benchmark file

Reviewer: ```suggestion
- Flags for `dumplog` and `force_logging_flush` were added to the C API
- Added missing C++98 call to `getCurrentTime`
- Added `closeLibrary` function to the C++98 API
- Added a Python benchmark file
```

- An option to install the benchmark executables has been added

> 

Reviewer: ```suggestion
- A GitHub Actions workflow to build packages for Linux with the benchmark executables

```
